NPE as MessagePublishProcessor tried to sendCorrelateCommand

**Describe the bug**

<!-- A clear and concise description of what the bug is. -->
A NullPointerException (NPE) was thrown while the `MessagePublishProcessor` tried to send the `CORRELATE` command. Specifically the [`flush`](https://github.com/camunda/zeebe/blob/bbbb0d3e8455efa051edde4a20e3152bc5e3636d/engine/src/main/java/io/camunda/zeebe/engine/processing/message/MessagePublishProcessor.java#L193) failed.

**To Reproduce**

<!--
Steps to reproduce the behavior

If possible add a minimal reproducer code sample
- when using the Java client: https://github.com/zeebe-io/zeebe-test-template-java

-->

Unknown.

**Expected behavior**

<!-- A clear and concise description of what you expected to happen. -->

Correlate command should be sent without an NPE.

**Log/Stacktrace**

<!-- If possible add the full stacktrace or Zeebe log which contains the issue. -->

[Full logs](https://drive.google.com/drive/folders/1vSP4GjEkMATh2ZaP84291E_M3-mCcmjp) (internal document)

<details><summary>Full Stacktrace</summary>
 <p>

```
java.lang.RuntimeException: java.lang.NullPointerException
	at io.camunda.zeebe.streamprocessor.DirectProcessingResult.executePostCommitTasks(DirectProcessingResult.java:69) ~[zeebe-workflow-engine-8.1.0-SNAPSHOT.jar:8.1.0-SNAPSHOT]
	at io.camunda.zeebe.streamprocessor.ProcessingStateMachine.lambda$executeSideEffects$11(ProcessingStateMachine.java:442) ~[zeebe-workflow-engine-8.1.0-SNAPSHOT.jar:8.1.0-SNAPSHOT]
	at io.camunda.zeebe.scheduler.retry.ActorRetryMechanism.run(ActorRetryMechanism.java:36) ~[zeebe-scheduler-8.1.0-SNAPSHOT.jar:8.1.0-SNAPSHOT]
	at io.camunda.zeebe.scheduler.retry.AbortableRetryStrategy.run(AbortableRetryStrategy.java:44) ~[zeebe-scheduler-8.1.0-SNAPSHOT.jar:8.1.0-SNAPSHOT]
	at io.camunda.zeebe.scheduler.ActorJob.invoke(ActorJob.java:99) ~[zeebe-scheduler-8.1.0-SNAPSHOT.jar:8.1.0-SNAPSHOT]
	at io.camunda.zeebe.scheduler.ActorJob.execute(ActorJob.java:47) ~[zeebe-scheduler-8.1.0-SNAPSHOT.jar:8.1.0-SNAPSHOT]
	at io.camunda.zeebe.scheduler.ActorTask.execute(ActorTask.java:120) ~[zeebe-scheduler-8.1.0-SNAPSHOT.jar:8.1.0-SNAPSHOT]
	at io.camunda.zeebe.scheduler.ActorThread.executeCurrentTask(ActorThread.java:106) ~[zeebe-scheduler-8.1.0-SNAPSHOT.jar:8.1.0-SNAPSHOT]
	at io.camunda.zeebe.scheduler.ActorThread.doWork(ActorThread.java:87) ~[zeebe-scheduler-8.1.0-SNAPSHOT.jar:8.1.0-SNAPSHOT]
	at io.camunda.zeebe.scheduler.ActorThread.run(ActorThread.java:198) ~[zeebe-scheduler-8.1.0-SNAPSHOT.jar:8.1.0-SNAPSHOT]
Caused by: java.lang.NullPointerException
	at java.util.Objects.requireNonNull(Unknown Source) ~[?:?]
	at io.camunda.zeebe.broker.transport.commandapi.CommandResponseWriterImpl.tryWriteResponse(CommandResponseWriterImpl.java:99) ~[zeebe-broker-8.1.0-SNAPSHOT.jar:8.1.0-SNAPSHOT]
	at io.camunda.zeebe.engine.processing.streamprocessor.writers.TypedResponseWriterImpl.flush(TypedResponseWriterImpl.java:114) ~[zeebe-workflow-engine-8.1.0-SNAPSHOT.jar:8.1.0-SNAPSHOT]
	at io.camunda.zeebe.engine.processing.message.MessagePublishProcessor.sendCorrelateCommand(MessagePublishProcessor.java:193) ~[zeebe-workflow-engine-8.1.0-SNAPSHOT.jar:8.1.0-SNAPSHOT]
	at io.camunda.zeebe.streamprocessor.DirectProcessingResult.executePostCommitTasks(DirectProcessingResult.java:67) ~[zeebe-workflow-engine-8.1.0-SNAPSHOT.jar:8.1.0-SNAPSHOT]
	... 9 more 
```

</p>
</details>

**Environment:**
- `ultrachaos`
- bbbb0d3e8455efa051edde4a20e3152bc5e3636d
- [reported error](https://console.cloud.google.com/errors/detail/CObOnMPDzIuWNA;service=zeebe;time=P7D?project=camunda-cloud-240911)

--- a/engine/src/main/java/io/camunda/zeebe/engine/processing/message/MessagePublishProcessor.java
+++ b/engine/src/main/java/io/camunda/zeebe/engine/processing/message/MessagePublishProcessor.java
@@ -177,19 +177,15 @@ public final class MessagePublishProcessor implements TypedRecordProcessor<Messa
   }
 
   private boolean sendCorrelateCommand() {
-
-    final var success =
-        correlatingSubscriptions.visitSubscriptions(
-            subscription ->
-                commandSender.correlateProcessMessageSubscription(
-                    subscription.getProcessInstanceKey(),
-                    subscription.getElementInstanceKey(),
-                    subscription.getBpmnProcessId(),
-                    messageRecord.getNameBuffer(),
-                    messageKey,
-                    messageRecord.getVariablesBuffer(),
-                    messageRecord.getCorrelationKeyBuffer()));
-
-    return success && responseWriter.flush();
+    return correlatingSubscriptions.visitSubscriptions(
+        subscription ->
+            commandSender.correlateProcessMessageSubscription(
+                subscription.getProcessInstanceKey(),
+                subscription.getElementInstanceKey(),
+                subscription.getBpmnProcessId(),
+                messageRecord.getNameBuffer(),
+                messageKey,
+                messageRecord.getVariablesBuffer(),
+                messageRecord.getCorrelationKeyBuffer()));
   }
 }

